torch/distributed/_tensor/dispatch.py

```diff
@@ -148,15 +150,111 @@ def operator_dispatch(
     suggested_input_schema = output_sharding.schema_suggestions[0]
     needs_redistribute = suggested_input_schema is not op_schema
 
+    def _arg_type_is_primitive(arg_type):
+        _python_primitive_types = [
+            "bool",
+            "int",
+            "float",
+            "double",
+        ]
+        return arg_type in _python_primitive_types
+
     if mesh is not None and mesh.get_coordinate() is None:
-        # if we are on a non-participating device, we simply return
-        # an empty tensor for now.
-        # TODO: what if the op returns a non-tensor value, what if
-        # the op returns a list of tensors, we need to figure out
-        # a consistent way to handle that, and also need to figure
-        # out if we should communicate the result to non-participating
-        # ranks (i.e. a.sum() -> scalar, maybe we should set to 0)
-        local_results = torch.tensor([])
+        # For a non-participating device, we do:
+        # 1. if the return type is scalar, all gather the local result
+        # from participating devices, and reduce on the list of results.
+        # For bool type, we by default use AND to reduce;
+        # For numeric type, we by default use ADD to reduce;
```

Comment:
I don't know if doing `add` for numeric type is correct or not. Could you give some example operator that behave like `add`? Otherwise I would say:
- if it's bool -> and
- else: raise RuntimeError

Reply:
You're right. Actually after checking `native_functions.yaml` I don't see any operator that returns numeric value and also dispatches to tensor computation.

Reply:
so we should delete it right? but it's still there

Reply:
Yes we should delete it. I haven't pushed the update to github yet.